src/MailAddress.php

```diff
@@ -7,21 +7,46 @@
 /**
  * @licence GNU GPL v2+
  * @author Christoph Fischer < christoph.fischer@wikimedia.de >
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
  */
 class MailAddress {
-	public $userName;
-	public $domain;
+	private $userName;
+	private $domain;
+	private $displayName;
 
-	public function __construct( string $userName, string $domain ) {
-		$this->userName = $userName;
-		$this->domain = $domain;
+	public function __construct( string $emailAddress, string $displayName = '' ) {
```

Comment:
I like you moved a bunch of stuff here. Not sure about the constructor change though. Might be better to have a `static newFromString` function. `$displayName` is not required by the existing users of this value object, which is an indication it does not belong here. How about creating a new value object, perhaps one that contains a MailAddress?